multiversx_sdk/abi/bytes_value.py

```diff
@@ -29,9 +27,20 @@ def decode_top_level(self, data: bytes):
     def set_payload(self, value: Any):
         if isinstance(value, str):
             self.value = bytes(value, "utf-8")
+        elif isinstance(value, dict):
+            value = cast(Dict[str, str], value)
```

Comment:
Maybe the cast is not necessary, since the linter sees an `Any` type passed to `_extract_value_from_dict` and is already forgiving?

Reply:
The cast is necessary since we have the `elif isinstance(value, dict)` statement.